Mathlib/NumberTheory/ModularForms/EisensteinSeries/Summable.lean

```diff
@@ -174,14 +177,26 @@ lemma linear_isTheta_left (d : ℤ) {z : ℂ} (hz : z ≠ 0) :
   · simp only [isLittleO_const_left, Int.cast_eq_zero,
       tendsto_norm_comp_cofinite_atTop_of_isClosedEmbedding Int.isClosedEmbedding_coe_real, or_true]
 
-lemma linear_inv_isBigO_right (c : ℤ) (z : ℂ) :
-    (fun (d : ℤ) ↦ (c * z + d)⁻¹) =O[cofinite] fun n ↦ (n : ℝ)⁻¹ :=
-  (linear_isTheta_right c z).inv.isBigO
+lemma linear_inv_isBigO_right_add (c e : ℤ) (z : ℂ) :
+    (fun (d : ℤ) ↦ (c * z + d + e)⁻¹) =O[cofinite] fun n ↦ (n : ℝ)⁻¹ :=
+  (linear_isTheta_right_add c e z).inv.isBigO
 
 lemma linear_inv_isBigO_left (d : ℤ) {z : ℂ} (hz : z ≠ 0) :
     (fun (c : ℤ) ↦ (c * z + d)⁻¹) =O[cofinite] fun n ↦ (n : ℝ)⁻¹ :=
   (linear_isTheta_left d hz).inv.isBigO
 
+lemma tendsto_zero_inv_linear (z : ℂ) (b : ℤ) :
+    Tendsto (fun d : ℕ ↦ 1 / ((b : ℂ) * z + d)) atTop (𝓝 0) := by
+  apply Asymptotics.IsBigO.trans_tendsto ?_ tendsto_inv_atTop_nhds_zero_nat (F'' := ℝ)
+  have := (Asymptotics.isBigO_sup.mp (Int.cofinite_eq ▸ linear_inv_isBigO_right_add b 0 z)).2
+  simpa [← Nat.map_cast_int_atTop, Asymptotics.isBigO_map] using this
+
+lemma tendsto_zero_inv_linear_sub (z : ℂ) (b : ℤ) :
+    Tendsto (fun d : ℕ ↦ 1 / ((b : ℂ) * z - d)) atTop (𝓝 0) := by
+  have := (tendsto_zero_inv_linear z (-b)).neg
+  simp only [Int.cast_neg, neg_mul, one_div, neg_zero, ← inv_neg] at *
+  exact this.congr (fun _ ↦ by ring)
```

Comment:
```suggestion
  grind [neg_zero, (tendsto_zero_inv_linear z (-b)).neg]
```
("grind" seems to be good when there are several distinct kinds of automation that need to be used together)